File: mobile/lib/features/channels/agent_activity/agent_activity_chrome.dart
/// Closed set of app-authored chrome strings for agent activity UI.
/// Keep byte-identical to desktop `agentActivityChrome.ts`.
abstract final class AgentActivityChrome {
  static const isWorking = 'is working';
  static String agentsWorking(int count) =>
      count == 1 ? '1 agent working' : '$count agents working';
  static const agentsWorkingLabel = 'Agents working';
  static const viewActivity = 'View activity';
  static const stop = 'Stop';
  static const seemsStuck = 'seems stuck';
  static const workingFallback = 'Working';
}

/// Hide the live activity line until the turn has been alive this long.
const activitySilenceMs = 3000;

/// Treat a turn as stuck when no new frame arrives for this long.
const activityStuckMs = 90000;

/// Format a live elapsed duration for a ticking counter.
/// Tiers: `<60s → "Ns"` · `<60m → "Nm Ns"` · `≥60m → "Nh Nm Ns"`.
String formatElapsed(int ms) {
  final totalSeconds = ms < 0 ? 0 : ms ~/ 1000;
  if (totalSeconds < 60) return '${totalSeconds}s';
  final seconds = totalSeconds % 60;
  final totalMinutes = totalSeconds ~/ 60;
  if (totalMinutes < 60) return '${totalMinutes}m ${seconds}s';
  final minutes = totalMinutes % 60;
  final hours = totalMinutes ~/ 60;
  return '${hours}h ${minutes}m ${seconds}s';
}

File: desktop/src/features/agents/ui/agentActivityChrome.ts
/**
 * Closed set of app-authored (chrome) strings for agent activity UI.
 * Content written by agents (checklist steps, questions) is never translated
 * here — keep that verbatim from the agent. Centralizing chrome makes a later
 * i18n pass a single-file change.
 */
export const AGENT_ACTIVITY_CHROME = {
  isWorking: "is working",
  agentsWorking: (count: number) =>
    count === 1 ? "1 agent working" : `${count} agents working`,
  agentsWorkingLabel: "Agents working",
  /** Counts runs, not agents: one agent can hold several live runs. */
  runsWorking: (count: number) => (count === 1 ? "1 run" : `${count} runs`),
  viewActivity: "View activity",
  openActivity: "Open Activity",
  /** Thread strip when no exact run identity is known: agent-scoped copy. */
  agentWorkingHint: (name: string) => `${name} is working — steer if stuck`,
  /** Agent-scoped Steer: focuses the composer, so it names the agent. */
  steerAgent: (name: string) => `Steer ${name}`,
  /** Agent-scoped Stop: stops the agent, not one run — the name says so. */
  stopAgent: (name: string) => `Stop ${name}`,
  steerRun: "Steer run",
  stopRun: "Stop run",
  sendSteer: "Send steer",
  /** Compact steer input: a name Activity's own input cannot collide with. */
  steerThisRunLabel: "Steer this run",
  /** Full Activity steer input. */
  steerSelectedRunLabel: "Steer selected run",
  stopSelectedRun: "Stop selected run",
  stop: "Stop",
  /** Thread-wide stop: names its scope so it cannot be read as one run. */
  stopAllRuns: "Stop all runs",
  /** Neutral trace left where the strip was after the operator stopped a run.
      Same words the Activity transcript uses for a cancelled turn. */
  runStopped: "Run stopped",
  seemsStuck: "seems stuck",
  workingFallback: "Working",
  retrying: (attempt: number, maxAttempts: number) =>
    `Retrying ${attempt}/${maxAttempts}`,
} as const;

/** Hide the live activity line until the turn has been alive this long. */
export const ACTIVITY_SILENCE_MS = 3_000;

/** Treat a turn as stuck when no new frame arrives for this long. */
export const ACTIVITY_STUCK_MS = 90_000;
